apps/ejabberd/src/mod_aws_sns.erl

```diff
@@ -161,6 +170,36 @@ user_presence_changed(#jid{lserver = Host} = UserJID, IsOnline) ->
 %%% Internal functions
 %%%===================================================================
 
+%% @doc Start publish process notification to AWS SNS service. Content should be valid JSON term
+-spec async_publish(Host :: ejabberd:lserver(), topic_arn(), Content :: jiffy:json_value(),
+              attributes()) -> ok.
+async_publish(Host, TopicARN, Content, Attributes) ->
+    Retry = opt(Host, publish_retry_count, 2),
+    wpool_worker:cast(?MODULE, ?MODULE, try_publish, [Host, TopicARN, Content, Attributes, Retry]).
```

Comment:
That would be hacky since I would have to somehow assume how wpool_worker API looks like. I can see that it uses messages like {M, F, A} but it feels like internal thing.